app/controllers/webhooks_controller.rb

```diff
@@ -49,11 +50,21 @@ def mailjet
   end
 
   def mandrill
-    events = params["mandrill_events"]
+    events = JSON.parse(params["mandrill_events"])
     events.each do |event|
       message_id = event.dig("msg", "metadata", "message_id")
       to_address = event.dig("msg", "email")
-      error_code = event.dig("msg", "diag")
+      error_string = event.dig("msg", "diag")
+      error_code = nil
+
+      if error_string.is_a? String
+        error_match = error_string.match(/\s\d\.\d\.\d\s/)
```

Comment:
I believe this regex is wrong. In this case, the SMTP code seems to be `550`, not `5.1.1`.

From [here](https://www.iana.org/assignments/smtp-enhanced-status-codes/smtp-enhanced-status-codes.xhtml) (find by "550"):

> The destination system exists and normally accepts mail, but something about the system has caused the generation of this DSN.

This is consistent with the description from the test.

Anyway, `bounce_error_code` is stored as a string, so you can probably just pass the whole error message as before.